src/medCore/medGlobal.cpp

```diff
@@ -0,0 +1,8 @@
+#include <medGlobal.h>
+#include <QCryptographicHash>
+
+QString anonymise(const QString name) {
+    QByteArray r = QCryptographicHash::hash(name.toUtf8(),QCryptographicHash::Md4).toHex().toUpper();
```

Comment:
Quick fix idea : maybe a loop checking if the hashed name is already in the DB, in which case we hash a second time (on the first hash) and repeat this until we are ok. On the very rare occasions where we have a clash, the second time around should be sufficient.